xonsh/pyghooks.py

```diff
@@ -44,6 +44,24 @@ class XonshLexer(PythonLexer):
         'subproc': SUBPROC_TOKENS,
     }
 
+    def get_tokens_unprocessed(self, text):
+        super_iter =  super().get_tokens_unprocessed(text)
+        # import pdb; pdb.set_trace()
+        #if not super_iter:
+        #    return super_iter
+        first=  next(super_iter)
+        from pygments.token import Token
+        from xonsh.built_ins import known_commands
+        if first:
+            if(first[1] == Token.Name) and (first[2] in known_commands()):
```

Comment:
Weird spacing here, too; you'd usually have a space between `if` and the `(`. (And you don't actually need the parentheses here—the operator precedence rules will do what you want—but having them for clarity makes sense.)